extensions/import/src/wizard/pages/prosePreviewPage.ts

```diff
@@ -73,27 +101,43 @@ export class ProsePreviewPage extends ImportPage {
 
 	public setupNavigationValidator() {
 		this.instance.registerNavigationValidator((info) => {
+			if (info) {
+				// Prose Preview to Modify Columns
+				if (info.lastPage === 1 && info.newPage === 2) {
+					return !this.loading.loading && this.table.data && this.table.data.length > 0;
+				}
+			}
 			return !this.loading.loading;
 		});
 	}
 
-	private async handleProse() {
-		await this.provider.sendPROSEDiscoveryRequest({
+	private async handleProse(): Promise<boolean> {
+		return await this.provider.sendPROSEDiscoveryRequest({
 			filePath: this.model.filePath,
 			tableName: this.model.table,
 			schemaName: this.model.schema,
 			fileType: this.model.fileType
 		}).then((result) => {
```

Comment:
try the onReject to see whether it is returning any errors. we should surface the error.

Reply:
This will need a backend change since this function never errors out

Reply:
The error would already be surfaced as an exception (and hopefully caught somewhere and displayed appropriately). 

You can get rid of the .then though - it's unnecessary with the await.

Reply:
There's no exception. I tried using a try-catch, but if there's a problem, the result is just null

Reply:
Would still suggest getting rid of the .then for readability though IMO

---
In reply to: [281312642](https://github.com/Microsoft/azuredatastudio/pull/5357#discussion_r281312642) [](ancestors = 281312642)

Reply:
We still need .then() to check if the result has null members or not.

Reply:
I'm saying take the code out of the then block and just have it directly after the await. The .then is just unnecessary code at this point.

```
let result = await this.provider.sendPROSEDiscoveryRequest(…)
if(result) { 
…
```

Reply:
I prefer using `then` for `Thenables` which is what `sendPROSEDiscoveryRequest` returns. That's standard and also helps in chaining promises. Ideally the backend will have a correct implementation, so we would have to switch to `then` anyway in this function since we don't get any errors here.

Reply:
Why? It's doing the exact same thing - but then is more verbose and makes debugging/error handling a lot harder.

https://hackernoon.com/6-reasons-why-javascripts-async-await-blows-promises-away-tutorial-c7ec10518dd9

And what does the return type have to do with it? .then returns a promise regardless. So whether we returned Thenable<boolean> or Promise<boolean> shouldn't affect the code inside.

Reply:
Our style going forward should be to always use async/await as far as I'm concerned - I don't know of any reason why we'd need to use the old Promises syntax. Refactoring the entire product to use that pattern now is too big but there's no reason we shouldn't clean this up as we go along.